.github/scripts/nno_dashboard/fetch_ci_data.py

```
@@ -0,0 +1,427 @@
+        results_by_ocp[ocp_version]["test_flavors"][test_flavor]["results"].append(result.to_dict())
+        results_by_ocp[ocp_version]["test_flavors"][test_flavor]["job_history_links"].add(job_history_url)
+
+        processed_count += 1
+
+    logger.info(f"Processed {processed_count} builds for PR #{pr_number}")
+
+
+def merge_and_save_results(
+    new_results: Dict[str, Dict[str, Any]],
+    output_filepath: str,
+    existing_results: Dict[str, Dict[str, Any]] = None
```

Ruff (0.15.15)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)